sites/public/src/components/listings/search/ListingsSearchForm.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from "react"
+import { ButtonGroup } from "./ButtonGroup"
+import { MultiSelectGroup } from "./MultiSelectGroup"
+import { ListingSearchParams, parseSearchString } from "../../../lib/listings/search"
+
+type ListingsSearchFormProps = {
+  searchString?: string
+  bedrooms: FormOption[]
+  bathrooms: FormOption[]
+  counties: FormOption[]
+  onSubmit: (params: ListingSearchParams) => void
+}
+
+export type FormOption = {
+  label: string
+  value: string
+  isDisabled?: boolean
+}
+
+const containerStyle: React.CSSProperties = {
+  width: "600px",
+  margin: "15px auto",
+  border: "1px solid black",
+}
+
+const headerStyle: React.CSSProperties = {
+  padding: "3px 15px",
+  borderBottom: "1px solid black",
+}
+
+const inputSectionStyle: React.CSSProperties = {
+  margin: "0px 15px",
+  borderTop: "1px solid black",
+}
+
+const textInputStyle: React.CSSProperties = {
+  border: "1px solid black",
+  padding: "2px 4px",
+  margin: "5px",
+}
+
+const footerStyle: React.CSSProperties = {
+  display: "flex",
+  padding: "3px 15px",
+  borderTop: "1px solid black",
+}
+
+const clearButtonStyle: React.CSSProperties = {
+  textDecoration: "underline",
+}
+
+/**
+ * A form for searching listings
+ *
+ * @param props
+ * @returns
+ */
+export function ListingsSearchForm(props: ListingsSearchFormProps) {
+  const searchString = props.searchString || ""
+
+  const nullState: ListingSearchParams = {
+    bedrooms: null,
+    bathrooms: null,
+    monthlyRent: "",
+    // Using an empty array here is how we know to look for an array value when parsing
+    counties: [],
+  }
+
+  const initialState = parseSearchString(searchString, nullState)
+
+  const [formValues, setFormValues] = useState(initialState)
+
+  const clearValues = () => {
```

Comment:
@lisunshiny This doesn't work and I have no idea why.  The state gets updated and the log statement executes, but the UI components don't change to reflect the empty state.  If you try to search again right after, it uses the empty state (ie no filters).  This is probably best addressed when the UI gets styled since the change in components will likely affect this anyway, but feel it's worth calling out